react/features/base/devices/functions.js

```diff
@@ -118,24 +146,29 @@ export function groupDevicesByKind(devices: Object[]): Object {
  * @param {string} newId - New audio output device id.
  * @param {Function} dispatch - The Redux dispatch function.
  * @param {boolean} userSelection - Whether this is a user selection update.
+ * @param {?string} newLabel - New audio output device label to store.
  * @returns {Promise}
  */
 export function setAudioOutputDeviceId(
         newId: string = 'default',
         dispatch: Function,
-        userSelection: boolean = false): Promise<*> {
+        userSelection: boolean = false,
+        newLabel: ?string): Promise<*> {
     return JitsiMeetJS.mediaDevices.setAudioOutputDevice(newId)
         .then(() => {
             const newSettings = {
                 audioOutputDeviceId: newId,
-                userSelectedAudioOutputDeviceId: undefined
+                userSelectedAudioOutputDeviceId: undefined,
+                userSelectedAudioOutputDeviceLabel: undefined
```

Comment:
Why does this need to be explicitly set to undefined?

Reply:
It is mentioned why a few lines below. This is flow workaround with the delete in the else below. Couldn't fix the flow error another way.
![image](https://user-images.githubusercontent.com/3263098/57158094-67021a80-6dda-11e9-932e-22abf9788290.png)